External/FEXCore/Source/Utils/Allocator.cpp

```diff
@@ -91,6 +92,44 @@ namespace FEXCore::Allocator {
     return Result;
   }
 
+  void *DisableSBRKAllocations() {
+    void* INVALID_PTR = reinterpret_cast<void*>(~0ULL);
+    // Get the starting sbrk pointer.
+    void *StartingSBRK = sbrk(0);
+    if (StartingSBRK == INVALID_PTR) {
+      // If sbrk is already returning invalid pointers then nothing to do here.
+      return INVALID_PTR;
+    }
+
+    // Now allocate the next page after the sbrk address to ensure it can't grow.
+    // In most cases at the start of `main` this will already be page aligned, which means subsequent `sbrk`
+    // calls won't allocate any memory through that.
+    void* AlignedBRK = reinterpret_cast<void*>(FEXCore::AlignUp(reinterpret_cast<uint64_t>(StartingSBRK), FHU::FEX_PAGE_SIZE));
+    void *AfterBRK = mmap(AlignedBRK, FHU::FEX_PAGE_SIZE, PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED_NOREPLACE | MAP_NORESERVE, -1, 0);
+    if (AfterBRK == INVALID_PTR) {
+      // Couldn't allocate the page after the aligned brk? This should never happen.
+      // FEXCore::LogMan isn't configured yet so we just need to print the message.
+      fextl::fmt::print("Couldn't allocate page after SBRK. Expect 32-bit thunks to break.\n");
```

Comment:
Seems like `ERROR_AND_DIE` is more suitable here, since it's a scenario we don't anticipate will happen. If it did, we'd want to notice quickly and see if we can do something about it.

Reply:
Can't use an `ERROR_AND_DIE` here since it is before LogManager is setup, but I did add a `FEX_TRAP_EXECUTION` just like `ERROR_AND_DIE` to behave the same but with output to stderr.